apps/ejabberd/src/ejabberd_sm.erl

```diff
@@ -686,14 +712,17 @@ do_route_offline(<<"iq">>, <<"error">>, _From, _To, _Packet) ->
 do_route_offline(<<"iq">>, <<"result">>, _From, _To, _Packet) ->
     ok;
 do_route_offline(<<"iq">>, _, From, To, Packet) ->
-    Err = jlib:make_error_reply(Packet, ?ERR_SERVICE_UNAVAILABLE),
-    ejabberd_router:route(To, From, Err);
+    El = mongoose_acc:to_element(Packet), % defensive programming, to be removed
```

Comment:
There is some mess, there are both accs and xmlels floating around and I'm not sure what to expect, so I do a kind of "safe conversion". Eventually all calls to to_element will be removed, as well as parts of code marked as ?DEPRECATED or ?TEMPORARY.